### ISSUE TITLE
Transfer body parameters from API Blueprint

### ISSUE THREAD

**Issue by mm-gmbd** [kminami/apib2swagger#9]
Currently, it doesn't look like the parameters (that are defined in the `+ Schema` block in API Blueprint) are transferred to body parameters in Swagger. Am I missing something?

    //API Blueprint Request Schema

    Some resource [GET]

    + Request
    
        +Schema
        
          {
            type: "object",
            properties: {
              "prop1": {
                type: "string",
                required: true
              }
            }
          }
      
        + Body
        
          {
            prop1: "a string"
          }

    //Swagger 2.0 
    {
      "/some/resource/":{
         "get": {
           ...
           "parameters": []
         } 
       }
    }

Also, by the Swagger 2.0 spec it looks like `parameters` is supposed to be an object, rather than array, it is supposed to be an [object of objects](http://swagger.io/specification/#parametersDefinitionsObject):

    {
      "skipParam": {
        "name": "skip",
        "in": "query",
        "description": "number of items to skip",
        "required": true,
        "type": "integer",
        "format": "int32"
      },
      "limitParam": {
        "name": "limit",
        "in": "query",
        "description": "max records to return",
        "required": true,
        "type": "integer",
        "format": "int32"
      }
    }

**Comment by kminami** [comment]
It is not implemented yet, because there are no normative words, references and examples that tell Schema section is described with JSON Schema on [current API Blueprint spesification](https://github.com/apiaryio/api-blueprint/blob/format-1A9/API%20Blueprint%20Specification.md#def-schema-section).

About Swagger 2.0, top level `parameters` is ParametersDefinitionsObject you said, but path item level `parameters` is [list of ParameterObjects](http://swagger.io/specification/#pathItemObject).
Currently, this project does not create top level one.

**Comment by mm-gmbd** [comment]
I'm not sure I 100% follow you. Considering the API Blueprint *schema* section is supposed to follow [JSON Schema](http://json-schema.org), then it, by default, has normative words (as well as references and examples). 

I agree that the actual API Blueprint documentation is lacking in that section (very badly, in fact), but there are several [JSON Schema examples](http://json-schema.org/examples.html) (another one on [Apiary here](https://docs.apiary.io/api_101/json-schema/)) out there.

I would think that, if there is a `Schema` definition under a `Request` definition, in API Blueprint, then the object under `Schema` would be easily translatable to body objects in Swagger (considering the API Blueprint `Schema` section must be valid JSON Schema).

**Comment by kminami** [comment]
OK, now I have implemented that.
Conversion example are [test/input/Schema.md](https://github.com/kminami/apib2swagger/blob/master/test/input/Schema.md) and [test/output/Schema.json](https://github.com/kminami/apib2swagger/blob/master/test/output/Schema.json).
Does it make sense for you?

**Comment by mm-gmbd** [comment]
@kminami, thanks for taking your time on this, however, I think it is a little off. 

Rather than automatically being encompassed in an object literally named `body`, if the API Blueprint `Schema` is of type `object`, the parameters listed in the `Schema` object (i.e. `prop1`), are inserted as individual parameters . For example:

# Some resource [POST /some/resource]

+ Request

    + Schema

            {
              "type": "object",
              "properties": {
                "prop1": {
                  "type": "string",
                  "required": true
                }
              }
            }

    + Body

            {
              "prop1": "a string"
            }

```
{
  "swagger": "2.0",
  "info": {
    "title": "",
    "version": "",
    "description": ""
  },
  "paths": {
    "/some/resource": {
      "post": {
        "responses": {},
        "summary": "Some resource",
        "description": "",
        "tags": [
          "(no tags)"
        ],
        "parameters": [
          {
            "name": "prop1",
            "in": "body",
            "schema": {
              "type": "string",
              "required": true
            }
          }
        ]
      }
    }
  },
  "definitions": {
    "Some resource": {},
  }
}
```

**Comment by kminami** [comment]
If there is a second parameter in the `Schema` object (i.e. `prop2`), do you mean the resulting parameters look like this?
```
"parameters": [ {
  "name": "prop1",
  "in": "body",
  "schema": {}
},{
  "name": "prop2",
  "in": "body",
  "schema": {}
}]
```

[Parameter Object section](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#parameterObject) says,
> Body - The payload that's appended to the HTTP request. Since there can only be one payload, there can only be one body parameter. The name of the body parameter has no effect on the parameter itself and is used for documentation purposes only. Since Form parameters are also in the payload, body and form parameters cannot exist together for the same operation.

So, I think above example is invalid.

**Comment by mm-gmbd** [comment]
Ah, indeed it sounds like my example would be invalid. I apologize for overlooking that specification language. Then, I would think that your original example would be correct. Thanks for your help! Has this already been merged into the master branch or is this potentially on a release branch?

**Comment by kminami** [comment]
Yes, already merged, and I have just published this to npm as v0.5.0. Thank you for supporting!
